packages/collections/range.pony

```diff
@@ -93,11 +93,12 @@ class Range[A: (Real[A] val & Number) = USize] is Iterator[A]
         or ((_min > _max) and (_inc > 0))
 
   fun has_next(): Bool =>
-    if _forward then
-      _idx < _max
-    else
-      _idx > _max
-    end
+    _infinite or
+      if _forward then
+        _idx < _max
+      else
+        _idx > _max
+      end
```

Comment:
I'd personally rather have this as two different if where it was something like:

```pony
if infinite 
  return true
end

if ...
```

@jemc and other @ponylang/committer, do you have thoughts or preference?

I find the way I suggested easier to quickly scan and read than the `_infinite or`.

Reply:
I also prefer the early return

Reply:
I changed it to your preferences.